blazor/datagrid/pdf-export.md

```diff
@@ -1059,6 +1059,107 @@ To apply theme in exported PDF, define the **theme** in export properties.
+    </GridColumns>
+</SfGrid>
+
+
+@code{
+    public SfGrid<Order> DefaultGrid;
+    public List<Order> Orders { get; set; }
+    List<string> headerValues = new List<string>();
+
+    public async Task ToolbarClickHandler(Syncfusion.Blazor.Navigations.ClickEventArgs args)
+    {
+        if (args.Item.Id == "Grid_pdfexport")  // Id is combination of Grid's ID and itemname.
+        {
+            PdfExportProperties ExportProperties = new PdfExportProperties();
+            ExportProperties.BeginCellLayout = new PdfGridBeginCellLayoutEventHandler(BeginCellEvent);
+            ExportProperties.FileName = "test.pdf";
+            ExportProperties.IsRepeatHeader = true;
+            await this.DefaultGrid.PdfExport(ExportProperties);
+        }
+    }
+
+    public void BeginCellEvent(object sender, PdfGridBeginCellLayoutEventArgs args)
+    {
+        PdfGrid grid = (PdfGrid)sender;
+        var brush = new Syncfusion.PdfExport.PdfSolidBrush(new Syncfusion.PdfExport.PdfColor(Color.DimGray));
+        args.Graphics.Save();
+        args.Graphics.TranslateTransform(args.Bounds.X + 50, args.Bounds.Height + 40);
+        args.Graphics.RotateTransform(-60);
+        // Draw the text at particular bounds.
+        args.Graphics.DrawString(headerValues[args.CellIndex], new Syncfusion.PdfExport.PdfStandardFont(Syncfusion.PdfExport.PdfFontFamily.Helvetica, 10), brush, new PointF(0, 0));
+        if (args.IsHeaderRow)
+        {
+            grid.Headers[0].Cells[args.CellIndex].Value = string.Empty;
+        }
+        args.Graphics.Restore();
+    }
+
+    public void PdfHeaderQueryCellInfoHandler(PdfHeaderQueryCellInfoEventArgs args)
+    {
+        headerValues.Add(args.Column.HeaderText);
+        var longestString = headerValues.Where(s => s.Length == headerValues.Max(m => m.Length)).
+                            First();
+        Syncfusion.PdfExport.PdfFont font = new Syncfusion.PdfExport.PdfStandardFont(Syncfusion.PdfExport.PdfFontFamily.Helvetica, 6);
+        SizeF size = font.MeasureString(longestString);
+        args.PdfGridColumn.Grid.Headers[0].Height = size.Width * 2;
+    }
+
+    protected override void OnInitialized()
+    {
+        Orders = Enumerable.Range(1, 75).Select(x => new Order()
+        {
+            OrderID = 1000 + x,
+            CustomerID = (new string[] { "ALFKI", "ANANTR", "ANTON", "BLONP", "BOLID" })[new Random().Next(5)],
+            Freight = 2.1 * x,
+            OrderDate = DateTime.Now.AddDays(-x),
+        }).ToList();
+    }
+
+    public class Order
+    {
+        public int? OrderID { get; set; }
+        public string CustomerID { get; set; }
+        public DateTime? OrderDate { get; set; }
+        public double? Freight { get; set; }
+    }
+}
+
+<style>
+    .e-grid .e-header {
+        font-weight: bold !important;
+    }
+
+    .e-grid .e-columnheader .e-headercell {
+        height: 100px;
+        transform: rotate(-60deg); // This is used to rotate the header text.
+    }
+</style> 
+```
+
```

Comment:
Please show the pdf grid exported image after this

Reply:
image added.